src/xmlsec/LegacyCertificate.py

```diff
@@ -0,0 +1,38 @@
+
+class LegacyCertificate(object):
```

Comment:
I called it legacy because it emulates the inner-api of pycrypto that was shown to the outside by us (and used at least by pyff). Shall I just add a calls to warn.warning or do we want something more sophisticated like pyff's decorator:

```
def deprecated(logger=log):
    """This is a decorator which can be used to mark functions
    as deprecated. It will result in a warning being emitted
    when the function is used."""

    def decorating(func):
        def new_func(*args, **kwargs):
            msg = "Call to deprecated function %s at %s:%d" % (func.__name__,
                                                               func.func_code.co_filename,
                                                               func.func_code.co_firstlineno + 1)
            if logger:
                logger.warn(msg)
            else:
                print(msg)

            return func(*args, **kwargs)

        return new_func
    return decorating
```

Reply:
I'd call it a shim and name it as such. If its legacy then there must be a clear path to removing the API but I don't really see that... or what am I missing?

Reply:
A deprecation warning serves two purposes:
1. indicate that the call will be removed in the future
2. suggest a migration path away from this call

The `@deprecated` decorator should include a _reason_ parameter.
The deprecation notice is also a _warning_ and should not bother with the logger.
Warnings can be optionally logged using [`logging.captureWarnings()`].

See also, the [deprecated] and [deprecation] packages as to how they handle this behaviour.

  [`logging.captureWarnings()`]: https://docs.python.org/3/library/logging.html#logging.captureWarnings
  [deprecated]: https://pypi.org/project/deprecated/
  [deprecation]: https://pypi.org/project/deprecation/

Reply:
@leifj Let's look at pem2cert: What do I want this function for?

a) parse a pem and tell me if it's correctly formed
b) give me some getters or attributes to access things like keysize, issuer etc.

At the moment this is done via a dictionary (fine) were one element is an object which itself used to be the backend cert object. Both things are used at the same time at least by pyff. I want to get rid of the exposure of the backend object but therefor we have to provide what's missing in the dictionary -- and I'm not sure if a dict is the right type of object for this job.

I'd say we create a new class (or 2) 'Certificate' and 'Key' that has getters for everything interesting. Again, c.f. #42 .

Reply:
The need for _getters only_ indicates that this is a _read-only_ object; iow it should be _immutable_. Another hint we get is that it only stores values. `dict` is not the right type because it allows its contents to change. To keep things simple, I would make this a [`namedtuple`] (which can also be exposed [as `dict`]); then revise if the needs turn out to be different.

  [`namedtuple`]: https://docs.python.org/3/library/collections.html#collections.namedtuple
  [as `dict`]: https://docs.python.org/3/library/collections.html#collections.somenamedtuple._asdict